dataprofiler/tests/profilers/profiler_options/test_histogram_option.py

```diff
@@ -0,0 +1,146 @@
+
+    def test_set(self):
+        option = self.get_options()
+
+        params_to_check = [
+            dict(prop='is_enabled', value_list=[False, True]),
+            dict(prop='method',
+                 value_list=['auto', 'fd', 'doane', 'scott', 'rice', 'sturges',
+                             'sqrt', ['sturges', 'doane']]),
+            dict(prop='hist_bin_count',
+                 value_list=['1', '10', '100', '1000', '99', '10000000'])
+        ]
+
+        # this code can be abstracted to limit code everywhere else
+        # AKA, params_to_check would be the only needed code plus raise errors
+        def _assert_set_helper(prop, value):
+            option.set({prop: value})
+            self.assertEqual(value, getattr(option, prop), msg=prop)
+
+        for params in params_to_check:
+            prop, value_list = params['prop'], params['value_list']
+            for value in value_list:
+                _assert_set_helper(prop, value)
+
+        # Treat method as a BooleanOption
+        expected_error = "type object 'method' has no attribute 'is_enabled'"
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'method.is_enabled': True})
+
+        # Treat hist_bin_count as a BooleanOption
+        expected_error = ("type object 'hist_bin_count' has no attribute "
+                          "'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'hist_bin_count.is_enabled': True})
+
+    def test_validate_helper(self):
+        super(TestHistogramOption, self).test_validate_helper()
+
+    def test_validate(self):
+
+        super(TestHistogramOption, self).test_validate()
+
+        params_to_check = [
+            # non errors
+            dict(prop='is_enabled', value_list=[False, True], errors=[]),
+            dict(prop='method',
+                 value_list=['auto', 'fd', 'doane', 'scott', 'rice', 'sturges',
+                             'sqrt', ['sturges', 'doane']],
+                 errors=[]),
+            dict(prop='hist_bin_count',
+                 value_list=[1, 10, 100, 1000, 99, 10000000],
+                 errors=[]),
+
+            # errors
+            dict(prop='method', value_list=[1, [], False],
+                 errors=[
+                     "HistogramOption.method must be a string or list of "
+                     "strings from the following: ['auto', 'fd', 'doane', "
+                     "'scott', 'rice', 'sturges', 'sqrt']."
+                 ]),
+            dict(prop='method', value_list=['whoops', ["doane", "incorrect"]],
+                 errors=[
+                     "HistogramOption.method must be a subset or selection "
+                     "from: ['auto', 'fd', 'doane', 'scott', 'rice', 'sturges',"
+                     " 'sqrt']."
+                 ]),
+            dict(prop='hist_bin_count', value_list=[-1, False, [], 1.2, 1.0],
```

Comment:
can we add a case with string, e.g., 'one', or '1'?

Reply:
adding